### ISSUE TITLE
Clarify white/grey/black-box is orthogonal to adaptive

### ISSUE THREAD

**Issue by carlini** [evaluating-adversarial-robustness/adv-eval-paper#23]
I don't want to try to define what "white/grey/black-box" means once-and-for-all, but I think it would be helpful to specifically say some things that different threat models are definitively *not*.

Specifically, it would be useful to document that the white/grey/black-box threat model is orthogonal to the adaptive/non-adaptive axis. There are adaptive black-box attacks (that try to do something special to break the defense in a black-box threat model) and non-adaptive white-box attacks (when you just run PGD un-modified).

The adaptive part is important even in the grey/black-box case. For example, adding even a tiny amount of randomness to the model will cause some black-box attacks (e.g., a naive implementation of the boundary attack) to fail completely. As a result, if the defense being studied is randomized, it would be necessary to adapt the boundary attack to that defense.

**Comment by wielandbrendel** [comment]
I totally agree. It's fruitless trying to exactly define whitebox vs grey-box vs black-box (they are all referring to different knowledge states of the attacker but where exactly people draw the boundary between those terms seems fuzzy and difficult to define on a more general level). But of course black-box attacks need to be adapted (e.g. by first averaging the decisions of a stochastic model before applying the boundary attack). I'd vote for including this statement in the manuscript.
